### DISCUSSION TITLE
What is randomized during simulate-batch with n_run = XX

### DISCUSSION THREAD

**Discussion by MartijnOW** [asreview/asreview#1214] [Q&A] [answered]
Hi ASReview community,

During my exploration of ASReview, I came across a few points that were not entirely clear to me. So, I would like to ask those in three separated discussions and I hope you can help me out! Just to explain the context: I want to perform a batch simulation (from the command prompt) with a number of runs, let's say 10. For that I used this code: asreview simulate-batch My_data_file.csv --state_file batchrun_test.h5 --init_seed 1 --seed 1 --n_run 10

In this example, I did not explicitly state: -e tfidf -m nb -q max -b double, as these should be the default settings.

I was wondering: what is exactly randomized when I perform a simulate-batch with n_run = XX and both initial seed and seed are set. Is it just the prior knowledge (prior_idx is not defined) that is varied or are the classifiers and feature extraction methods etc. also varied when not explicitly specified during a batch run? 

And as follow up question (and for my insight): does a batch run with n_run = 10 & both seeds set result in one of the seeds changing during each run (as I can see that at least the prior knowledge is changed for each run I suppose the init_seed is changed?)

**Comment by jteijema** [comment, marked as answer]
Hi @MartijnOW,

The [init_seed](https://asreview.readthedocs.io/en/latest/simulation_cli.html#cmdoption-init_seed) option sets the `prior_idx` seed. So this means that with `simulate-batch` all priors will be set to the same records. 

As for what is randomized, not a lot! If you're using the settings you selected, then the simulations settings are the same for each run. However, not all models are deterministic. Any neural network or parallel processed model is inherently different with each run, even when a seed is set.


`Simulate-batch` is a feature of the older v0.19.x of ASReview, and was removed because, like you found, settings behave differently from `simulate` to `simulate-batch`. For the new v1.0, we created ASReview's [`Makita`](https://github.com/asreview/asreview-makita) (Make It Automatic). We developed this package especially for running multiple simulations. It gives full control over every separate simulation and makes it fully reproducible and adjustable to your needs.

Using v1.0 and the `Makita` package, running `asreview Makita template basic -f jobs.bat --n_runs 10 --init_seed 1 --model_seed 1` creates a script with 10 simulations, and running this script will achieve the same as `simulate-batch` would do, but then in a reproducible and clear way. For instructions on how to use Makita, check the page [readme](https://github.com/asreview/asreview-makita)

Don't be afraid to ask anything about the usage on the Makita discussion page!
Jelle

**Reply by MartijnOW** [reply]
Hi Jelle, 

This is great. Thank you so much. This tool looks great.
Unfortunately, there is an issue that keeps popping up when I try to run the Makita tool. I followed the steps on the Read.me file. But when I then run the lines of the makita command (like "asreview makita template basic -f jobs.bat --n_runs 10") I keep getting the following error directly after the script has checked my labelled datasets (by n_records, n_relevant, n_irrelevant, n_unlabeled, n_missing title, n_missing_abstract, and n_duplicates): 

"WARNING:root:Plugin with entry point stat could not be loaded.
usage: asreview [-h] [-V] [subcommand]" 

I am fairly sure that this means that there's something wrong with the pathfinding, although I am a 100% sure that I created the correct pathway to the project folder (using cd C:\Users\etc\etc\etc). Do you have any thought on what could cause this error every time I run the Makita tool? 

I have upgrated my ASReview and Makita to the newest version.

I don't think there is a discussion page yet on the Makita github folder, so hopefully it is okay to leave my question here.

Kind regards,
Martijn

**Reply by Rensvandeschoot** [reply]
can you try [upgrading ](https://asreview.readthedocs.io/en/latest/installation.html#upgrade-asreview) to v1.0.3?

**Reply by jteijema** [reply]
Hi @MartijnOW 

What I think is happening, is that not all asreview extensions are installed. The script makes use of asreview Datatools, Insights and Wordcloud. I will make sure to update the documentation with that information!

The fix is pretty easy for you. You can either remove some lines from the script file (I think the error is resulting from the Wordcloud generation) or install the packages.

Wordcloud van be installed with `pip install asreview-wordcloud`, Insights with `pip install asreview-insights` and data tools with `pip install asreview-datatools`. 

Thank you for using this package! User feedback like this is very important for the development of packages like this.

Jelle
